lib/hets-agent/hets/analysis_caller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'hets-agent/hets/caller'
+
+module HetsAgent
+  module Hets
+    # Provides an interface to call Hets and analyze a Document
+    class AnalysisCaller < Caller
```

Comment:
Seems more like a `AnalysisRequest` to me, than a caller (I'd expect a caller to be more or less stateless, at least in respect to a specific request)

Reply:
Maybe we could split the instantiation of a request (`initialize`) from the actual caller (the rest of the class, with `call` receiving an instance of the request)

Reply:
Then, we have a problem with differing interfaces. VersionCaller returns a String while the return value of AnalysisCaller is a Boolean. What should we expect the Caller to return?

Reply:
Done in d61af44.

Reply:
Done in 4fd07db